### ISSUE TITLE
suggestion: grouping: support automatic pinning

### ISSUE THREAD

**Issue by janppires** [angular-ui/ui-grid#4123]
It would be great to automatically pin a column when grouping by the column.

I've tried to implement a workaround with the available gridApi, but I cannot find an easy way to see if the column is already pinned (for example, the existance of an isPinnedLeft attribute) or already grouped.

pseudo-code:
    gridApi.grouping.on.groupingChanged(col){
        if(!col.isPinnedLeft){
            gridApi.pinning.pinColumn(col, uiGridPinningConstants.container.LEFT)
        }
    ...
    }

**Comment by PaulL1** [comment]
You shouldn't need to pin, it automatically moves the column to the left.  I had reason at the time not to use pinning - I think it was that I'd then have a dependency on the pinning feature that I didn't want.

**Comment by janppires** [comment]
Yes, it automatically moves the column to the left! But, if you need to have a table with horizontal scroll, that feature could be very usefull. I'm having that use case! =)

**Comment by janppires** [comment]
I'm OK about implementing myself the logic to get that behavior. I only need gridApi with some methods that can help me achieve that. thank you!

**Comment by JLLeitschuh** [comment]
If you have the code please post it here so others can use it in their own projects.

**Comment by janppires** [comment]
I don't have any code since the gridApi does not support an easy way to implement the behavior needed. I will be glad to contribute my code after that. =)
